app/scripts/modules/core/src/pipeline/config/stages/deployService/deployServiceStage.ts

```diff
@@ -0,0 +1,18 @@
+import { module } from 'angular';
+
+import { Registry } from 'core/registry';
+import { ExecutionDetailsTasks } from '../core';
+import { DeployServiceExecutionDetails } from './DeployServiceExecutionDetails';
+
+export const DEPLOY_SERVICE_STAGE = 'spinnaker.core.pipeline.stage.deployService';
+
+module(DEPLOY_SERVICE_STAGE, []).config(() => {
+  Registry.pipeline.registerStage({
```

Comment:
Same concern here regarding stage visibility with Cloudfoundry disabled.

Reply:
It does not show up if you don't enable Cloud Foundry.

The property I think you are referring to is `useBaseProvider: true`, which is configured.